### PR TITLE
[10.0][MIG] partner_financial_risk and partner_sale_risk

### PR DESCRIPTION
**Pull request by Ehtaga**

### CODE REVIEW

File: partner_financial_risk/models/account_invoice.py
Comment (chienandalu) [inline partner_financial_risk/models/account_invoice.py]: Although the logic is the same is better to minimize nesting if possible.

**Review by chienandalu — COMMENTED** [review]
Hi, @Ehtaga. Thanks for your work! Some quick comments:

- Is a good practice to make a PR for module migration. Reviews are easier and commit history is easier to follow as well.
- Fixup transifex commits to have the minimum ones.

**Review by chienandalu — CHANGES_REQUESTED** [review]
Seems also that your tests are failing on Travis :confused:

**Comment by Ehtaga** [comment]
@chienandalu The tests are failing on Travis but not in local, and not on our Jenkins. I think there is a problem with Travis (the error is in the common tests `setUp()`), so we can't spend more time on it.

I don't understand your comment about the Transifex commits. Do you want me to amend them in a single commit? I don't think it is a good idea as we would loose trace of what changed and when it has been done.

About your nesting comment, this is a migration, so I didn't change the functioning code. I personally think that this piece of code is ok. It is a `@multi` method so the `for` is unavoidable. There are only 2 levels of `if`, so it wouldn't help that much to refactor it.

**Comment by pedrobaeza** [comment]
@Ehtaga, we don't need to keep the bot commits trace, and less in a migration, so you can squash all of them together. See https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

File: README.md
Comment (pedrobaeza) [inline README.md:43]: This shouldn't be changed here. It's changed automatically. Please remove any trace of this or it will conflict as soon as other module is merged in the repository.

**Comment by pedrobaeza** [comment]
You have to split this PR in 2: let here the main module, and make another with the secondary module.

File: partner_financial_risk/tests/test_partner_financial_risk.py
Comment (pedrobaeza) [inline partner_financial_risk/tests/test_partner_financial_risk.py:9]: Don't depend on AccountingTestCase but directly on TransactionCase and this will work on Travis. Take into account that the test on Travis are not performed the same as in local: first the module is installed, and then Odoo is re-run executing the tests.

Comment (Ehtaga) [reply partner_financial_risk/tests/test_partner_financial_risk.py:9]: We chose to depend on AccountingTestCase because there was a bug with the traditional one. The tests were asking to install a chart of account:
```
2017-06-28 20:08:48,434 13918 INFO MaisonDF-Tests odoo.modules.module: odoo.addons.partner_sale_risk.tests.test_partner_sale_risk running tests.
2017-06-28 20:08:49,840 13918 INFO MaisonDF-Tests odoo.addons.partner_sale_risk.tests.test_partner_sale_risk: test_invoice_amount (odoo.addons.partner_sale_risk.tests.test_partner_sale_risk.TestPartnerSaleRisk)
2017-06-28 20:08:51,074 13918 ERROR MaisonDF-Tests odoo.addons.partner_sale_risk.tests.test_partner_sale_risk: ERROR
2017-06-28 20:08:51,081 13918 INFO MaisonDF-Tests odoo.addons.partner_sale_risk.tests.test_partner_sale_risk: test_sale_order (odoo.addons.partner_sale_risk.tests.test_partner_sale_risk.TestPartnerSaleRisk)
2017-06-28 20:08:52,524 13918 INFO MaisonDF-Tests odoo.addons.partner_sale_risk.tests.test_partner_sale_risk: ======================================================================
2017-06-28 20:08:52,524 13918 ERROR MaisonDF-Tests odoo.addons.partner_sale_risk.tests.test_partner_sale_risk: ERROR: test_invoice_amount (odoo.addons.partner_sale_risk.tests.test_partner_sale_risk.TestPartnerSaleRisk)
2017-06-28 20:08:52,524 13918 ERROR MaisonDF-Tests odoo.addons.partner_sale_risk.tests.test_partner_sale_risk: Traceback (most recent call last):
2017-06-28 20:08:52,525 13918 ERROR MaisonDF-Tests odoo.addons.partner_sale_risk.tests.test_partner_sale_risk: `   File "/var/lib/jenkins/workspace/MaisonDF-Tests/git/partner-contact/partner_sale_risk/tests/test_partner_sale_risk.py", line 69, in test_invoice_amount
2017-06-28 20:08:52,525 13918 ERROR MaisonDF-Tests odoo.addons.partner_sale_risk.tests.test_partner_sale_risk: `     wizard.create_invoices()
2017-06-28 20:08:52,525 13918 ERROR MaisonDF-Tests odoo.addons.partner_sale_risk.tests.test_partner_sale_risk: `   File "/var/lib/jenkins/workspace/MaisonDF-Tests/git/odoo/addons/sale/wizard/sale_make_invoice_advance.py", line 164, in create_invoices
2017-06-28 20:08:52,525 13918 ERROR MaisonDF-Tests odoo.addons.partner_sale_risk.tests.test_partner_sale_risk: `     self._create_invoice(order, so_line, amount)
2017-06-28 20:08:52,526 13918 ERROR MaisonDF-Tests odoo.addons.partner_sale_risk.tests.test_partner_sale_risk: `   File "/var/lib/jenkins/workspace/MaisonDF-Tests/git/odoo/addons/sale/wizard/sale_make_invoice_advance.py", line 75, in _create_invoice
2017-06-28 20:08:52,526 13918 ERROR MaisonDF-Tests odoo.addons.partner_sale_risk.tests.test_partner_sale_risk: `     (self.product_id.name,))
2017-06-28 20:08:52,526 13918 ERROR MaisonDF-Tests odoo.addons.partner_sale_risk.tests.test_partner_sale_risk: ` UserError: (u'There is no income account defined for this product: "Down payment". You may have to install a chart of account from Accounting app, settings menu.', None)
2017-06-28 20:08:52,526 13918 INFO MaisonDF-Tests odoo.addons.partner_sale_risk.tests.test_partner_sale_risk: Ran 2 tests in 4.089s
2017-06-28 20:08:52,527 13918 ERROR MaisonDF-Tests odoo.addons.partner_sale_risk.tests.test_partner_sale_risk: FAILED
2017-06-28 20:08:52,527 13918 INFO MaisonDF-Tests odoo.addons.partner_sale_risk.tests.test_partner_sale_risk:  (errors=1)
2017-06-28 20:08:52,527 13918 ERROR MaisonDF-Tests odoo.modules.module: Module partner_sale_risk: 0 failures, 1 errors
```
The problem was solved by depending on AccountingTestCase. And it worked with our Jenkins, so I won't have time to find another solution.

**Comment by Ehtaga** [comment]
@pedrobaeza I can separate this in 2 PR and clean the `README.md` but I can't spend more time on this migration as it is perfectly working with our own Jenkins.

**Comment by pedrobaeza** [comment]
@Ehtaga sorry, but "working perfectly" on your system doesn't mean it is going to work for all the environments (including Travis). That only reveals an underlying problem. You have admitted the change on the inheritance because previously there was a problem, and the change you have made only masks the situation, as the tests are not performed if there isn't any chart of accounts installed. That's why I think your Jenkins CI is not failing. I encourage you to please check again and let this working correctly (splitting the PRs for better granularity).

**Comment by pedrobaeza** [comment]
Superseeded by #470 and #452

**Pull request closed without merging**
